packages/form-js-viewer/src/util/simple.js

```diff
@@ -48,18 +48,22 @@ export function clone(data, replacer) {
 }
 
 /**
- * Wrap an expression context with additional local context.
- * A version of the local context with underscore-wrapped keys is also injected as fallback.
+ * Transform a LocalExpressionContext object into a usable FEEL context.
  *
- * @param {Object} baseContext - The context to wrap.
- * @param {Object} localContext - The local context object.
- * @returns {Object} The merged context object.
+ * @param {Object} context - The LocalExpressionContext object.
+ * @returns {Object} The usable FEEL context.
  */
-export function wrapExpressionContext(baseContext, localContext) {
+
+export function buildExpressionContext(context) {
+  const {
+    data,
+    ...specialContextKeys
```

Comment:
All keys which aren't data, basically they need to be accessible via a fallback in case the data itself contains that key.

Example:

```
{
   value1: "a"
   group:  {
     parent: {
       value1: "b"
     }
   }
}
```

In this context, the variable parent exists within the group, so we provide a fallback value called _parent_ which is a way to allow the value to be both used as a local variable, or if there's ambiguity defining it with the _parent_ variable which will always overwrite no matter what. This does mean that we always overwrite _parent_, but that's also why we make that variable a little unique with the wrapping.